Bottom: 1a84d1c2a473c78c80ce4742201894980dca1dfa
Top:    3abbaac932ac04ac1113129ab0363cb0018921eb
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-10-22 05:41:17 +0100

Attempt some GL optimisation


---

diff --git a/src/hid/common/hidgl.c b/src/hid/common/hidgl.c
index 54c2b3b..4489da4 100644
--- a/src/hid/common/hidgl.c
+++ b/src/hid/common/hidgl.c
@@ -214,7 +214,7 @@ static void draw_cap (double width, int x, int y, double angle, double scale)
 void
 hidgl_draw_line (int cap, double width, int x1, int y1, int x2, int y2, double scale)
 {
-  double angle;
+  double angle = 0.;
   float deltax, deltay, length;
   float wdx, wdy;
   int circular_caps = 0;
@@ -243,10 +243,15 @@ hidgl_draw_line (int cap, double width, int x1, int y1, int x2, int y2, double s
     wdy = deltax * width / 2. / length;
     wdx = -deltay * width / 2. / length;
 
-    if (deltay == 0.)
-      angle = (deltax < 0) ? 270. : 90.;
+    if (hairline)
+      angle = 0;
     else
-      angle = 180. / M_PI * atanl (deltax / deltay);
+      {
+        if (deltay == 0.)
+          angle = (deltax < 0) ? 270. : 90.;
+        else
+          angle = 180. / M_PI * atanl (deltax / deltay);
+      }
 
     if (deltay < 0)
       angle += 180.;
diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index 5ab4162..5e4510b 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -1680,6 +1680,7 @@ ghid_port_drawing_area_expose_event_cb (GtkWidget * widget,
   glEnable (GL_BLEND);
   glBlendFunc (GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
 
+  glShadeModel (GL_FLAT);
 //  glEnable(GL_POLYGON_SMOOTH);
 //  glHint(GL_POLYGON_SMOOTH_HINT, [GL_FASTEST, GL_NICEST, or GL_DONT_CARE]);
